isolated-jobs-test/src/main/java/com/hazelcast/jet/tests/isolated/IsolatedJobsStreamTest.java

```diff
@@ -0,0 +1,170 @@
+import com.hazelcast.jet.pipeline.Sinks;
+import com.hazelcast.jet.pipeline.WindowDefinition;
+import com.hazelcast.jet.tests.common.AbstractSoakTest;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+import static com.hazelcast.jet.config.ProcessingGuarantee.AT_LEAST_ONCE;
+import static com.hazelcast.jet.core.JobStatus.RUNNING;
+import static com.hazelcast.jet.tests.common.Util.getJobStatusWithRetry;
+import static com.hazelcast.jet.tests.common.Util.sleepMinutes;
+import static com.hazelcast.jet.tests.common.Util.waitForJobStatus;
+import static com.hazelcast.jet.tests.isolated.JetMemberSelectorUtil.excludeMember;
+import static java.util.stream.Collectors.joining;
+
+/**
+ * This test requires Enterprise License with Advanced Compute Feature.
+ * To run this test locally, you have to use EE version (it can be forced by adding dependency to hazelcast-enterprise
+ * in the pom.xml) and license must be set (it can be done in {@link AbstractSoakTest#run(String[])})
+ */
+public class IsolatedJobsStreamTest extends AbstractSoakTest {
+    private static final int DEFAULT_LOG_VERIFICATION_COUNT_THRESHOLD = 5;
+    private static final int DEFAULT_SLEEP_BETWEEN_VALIDATIONS_IN_MINUTES = 2;
+    private static final int DEFAULT_JOB_WINDOW_TUMBLE_IN_MINUTES = 1;
+    private static final int DEFAULT_SNAPSHOT_INTERVAL_MS = 1000;
+    private static final int DEFAULT_SLEEP_BETWEEN_STREAM_RECORDS_MS = 1000 * 30;
+    private int snapshotIntervalMs;
+    private int logVerificationCountThreshold;
+    private int sleepBeforeValidationInMinutes;
+    private int jobWindowTumbleInMinutes;
+    private int sleepBetweenStreamRecordsMs;
+
+    public static void main(final String[] args) throws Exception {
+        new IsolatedJobsStreamTest().run(args);
+    }
+
+    @Override
+    protected boolean runOnBothClusters() {
+        return true;
+    }
+
+    @Override
+    protected void init(HazelcastInstance client) throws Exception {
+        snapshotIntervalMs = propertyInt("snapshotIntervalMs",
+                DEFAULT_SNAPSHOT_INTERVAL_MS);
+        logVerificationCountThreshold = propertyInt("logVerificationCountThreshold",
+                DEFAULT_LOG_VERIFICATION_COUNT_THRESHOLD);
+        sleepBeforeValidationInMinutes = propertyInt("sleepBeforeValidationInMinutes",
+                DEFAULT_SLEEP_BETWEEN_VALIDATIONS_IN_MINUTES);
+        jobWindowTumbleInMinutes = propertyInt("jobWindowTumbleInMinutes",
+                DEFAULT_JOB_WINDOW_TUMBLE_IN_MINUTES);
+        sleepBetweenStreamRecordsMs = propertyInt("sleepBetweenStreamRecordsMs",
+                DEFAULT_SLEEP_BETWEEN_STREAM_RECORDS_MS);
+    }
+
+    @Override
+    protected void test(HazelcastInstance client, String name) throws Throwable {
+        long begin = System.currentTimeMillis();
+        String logPrefix = "[" + name + "] ";
+
+        UUID excludedMember = client.getCluster().getMembers().iterator().next().getUuid();
```

Comment:
Member UUID is changed after the restart, right? I means that after the first round of restarts we actually doesn't check anything and run jobs on all members (since none of members is `excludedMember`).

I think we should rather use `member-attributes` for this filtering (or something else, but UUID is not a good idea).

Reply:
Changed to network address.